### PR TITLE
Nginx and node improvements

### PR DESCRIPTION
**Pull request by clint-tseng**
1. Enable gzip on nginx.
2. Always rebuild the nginx conf file so it picks up conf changes.
3. Start using multithreaded node.

### CODE REVIEW

File: files/nginx/odk.conf.template [hunk 1]
```diff
@@ -15,6 +15,12 @@ server {
 
   client_max_body_size 100m;
 
+  gzip on;
+  gzip_vary on;
+  gzip_min_length 1280;
```
Comment (yanokwa) [inline files/nginx/odk.conf.template]: I've seen 1000 on the interwebs. Why did you choose 1280 bytes?

File: files/service/scripts/start-odk.sh [hunk 2]
```diff
@@ -12,5 +12,6 @@ echo "starting cron.."
 cron -f &
 
 echo "starting server."
-node lib/bin/run-server.js
+mkdir -p /var/log/odk
+node node_modules/naught/lib/main.js start --remove-old-ipc true --worker-count 4 --daemon-mode false --log /var/log/odk/naught.log --stdout /var/log/odk/stdout.log --stderr /var/log/odk/stderr.log lib/bin/run-server.js
```
Comment (yanokwa) [inline files/service/scripts/start-odk.sh]: Why 4 workers?

File: files/nginx/odk.conf.template (hunk 1 above)
Comment (clint-tseng) [reply files/nginx/odk.conf.template]: _shrug_. It's not a terribly sensitive value, and the thing I copied had 1280 and 2560 as suggestions. I'd comment on it, but the comment would simply read "why not?" Honestly some of this stuff is as much an incantation to me as it is to you. There is definite cargo-culting in this block of config.

**Comment by yanokwa** [comment]
Thanks for the clarifications. I just like to have some record of where the magic numbers come from.

Merge whenever you are ready!

**Pull request merged**